.gitlab/wait-for-service-ready.sh

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+HOST=$1
+PORT=$2
+SERVICE_TYPE=${3:-generic}
+MAX_ATTEMPTS=${4:-30}
+RETRY_DELAY=${5:-5}
+
+echo "Waiting for $HOST:$PORT to be reachable..."
+wait-for "$HOST:$PORT" --timeout=180
```

Comment:
```suggestion
if ! wait-for "$HOST:$PORT" --timeout=180; then
    echo "ERROR: Could not reach $HOST:$PORT" >&2
    exit 1
fi
```
Should we early exit here?